### ISSUE TITLE
kubeadm should have an option on master to print out join command again

### ISSUE THREAD

**Issue by peterkellyonline** [kubernetes/kubernetes#34324]
**What keywords did you search in Kubernetes issues before filing this one?** (If you have found any duplicates, you should instead reply there.):

kubeadm master join

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):

FEATURE REQUEST 

A convenient command such as `kubeadm config` or `kubeadm show` to return the command for a node to join the cluster i.e. `kubeadm join --token <token> <master-ip>`. 

(I am happy to work on this myself and contribute)

**Comment by luxas** [comment]
I'd say feel free to :)
cc @kubernetes/sig-cluster-lifecycle

**Comment by errordeveloper** [comment]
I think you might be able to solve this with `kubeadm init --token=...`. If you want to be more specific about the IP address as well, you can pass `--api-advertise-addresses` as well. We have advance usage docs here now: http://kubernetes.io/docs/admin/kubeadm/.

**Comment by luxas** [comment]
I think the use case was to let kubeadm generate a random token and be able to fetch it later, but I don't know if it's worthwhile... May be easier to just specify a random token to kubeadm init directly

**Comment by peterkellyonline** [comment]
Yes, the use case letting init generate the random token but then be able
to fetch that token later, outside of the print out from the successful run
of init on master node.

It is straight forward to get it through the master API, then decode the
value to JSON and get the token that way, but I expected to be able to get
this easily from kubeadm cli.

If it's not worthwhile and the advice is to generate and store your own
token first, before init, then that's okay.

**Comment by pires** [comment]
I believe this is something @dgoodwin is working on.

**Comment by dgoodwin** [comment]
Yes haven't been able to start but I'm hoping to start designing richer token management.

**Comment by pires** [comment]
@dgoodwin if you're good with it, I can implement this simple functionality and then when token mgmt becomes a thing I'm ok with replacing it.

**Comment by pires** [comment]
@luxas please assign to me.

**Comment by luxas** [comment]
FYI, you can fetch the token already this way:
```bash
kubectl -n kube-system get secret clusterinfo -o yaml | grep token-map | awk '{print $2}' | base64 -d | sed "s|{||g;s|}||g;s|:|.|g;s/\"//g;" | xargs echo
```

May be a bit hacky, but it works :smile:

**Comment by luxas** [comment]
@dgoodwin I guess we could come up with something like `kubeadm token list -n 1` for this

**Comment by dgoodwin** [comment]
Yeah token list is on my "list".

**Comment by pires** [comment]
@luxas then please re-assign to @dgoodwin.

**Comment by dgoodwin** [comment]
Fixed in #35805.

**Comment by mikedanese** [comment]
This issue was moved to kubernetes/kubeadm#26
